containerd/containerd.go

```diff
@@ -342,6 +380,7 @@ func (d *Driver) createContainer(containerConfig *ContainerConfig, config *TaskC
 		containerd.WithRuntime(d.config.ContainerdRuntime, nil),
 		containerd.WithNewSnapshot(containerConfig.ContainerSnapshotName, containerConfig.Image),
 		containerd.WithNewSpec(opts...),
+		withFileLimit(uint64(config.FileLimit)),
```

Comment:
Instead of unmarshalling just-built spec back, I'd suggest setting rlimit by appending your function to `opts`

Reply:
Thanks for pointing that out, I had to read a load of source code to find the right knobs to twist for this and that was something I missed. New revision just pushed up and verified in my test system!